iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -137,26 +137,12 @@ private static Table loadTable(Configuration conf, String tableIdentifier, Strin
    * @return the created Iceberg table
    */
   public static Table createTable(Configuration conf, Properties props) {
-    String schemaString = props.getProperty(InputFormatConfig.TABLE_SCHEMA);
-    Preconditions.checkNotNull(schemaString, "Table schema not set");
-    Schema schema = SchemaParser.fromJson(props.getProperty(InputFormatConfig.TABLE_SCHEMA));
-
-    String specString = props.getProperty(InputFormatConfig.PARTITION_SPEC);
-    PartitionSpec spec = PartitionSpec.unpartitioned();
-    if (specString != null) {
-      spec = PartitionSpecParser.fromJson(schema, specString);
-    }
-
+    Schema schema = getSchema(props);
+    PartitionSpec spec = getPartitionSpec(props, schema);
     String location = props.getProperty(LOCATION);
     String catalogName = props.getProperty(InputFormatConfig.CATALOG_NAME);
 
-    // Create a table property map without the controlling properties
-    Map<String, String> map = Maps.newHashMapWithExpectedSize(props.size());
-    for (Object key : props.keySet()) {
-      if (!PROPERTIES_TO_REMOVE.contains(key)) {
-        map.put(key.toString(), props.get(key).toString());
-      }
-    }
+    Map<String, String> map = getIcebergTableProperties(props);
```

Comment:
nit:filterIcebergTableProperties?
We do not use `get` in Iceberg code path, so we conform the Iceberg coding standards